File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/entities/dtos/HouseholdAccountShareDto.java
package com.decrux.pocketr.api.entities.dtos;

import java.time.Instant;
import java.util.UUID;

public record HouseholdAccountShareDto(
    UUID accountId,
    String accountName,
    String ownerEmail,
    String ownerFirstName,
    String ownerLastName,
    Instant sharedAt
) {
    public UUID getAccountId() {
        return accountId;
    }

    public String getAccountName() {
        return accountName;
    }

    public String getOwnerEmail() {
        return ownerEmail;
    }

    public String getOwnerFirstName() {
        return ownerFirstName;
    }

    public String getOwnerLastName() {
        return ownerLastName;
    }

    public Instant getSharedAt() {
        return sharedAt;
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr_api/entities/dtos/HouseholdAccountShareDto.kt
package com.decrux.pocketr_api.entities.dtos

import java.time.Instant
import java.util.UUID

data class HouseholdAccountShareDto(
    val accountId: UUID,
    val accountName: String,
    val ownerEmail: String,
    val ownerFirstName: String?,
    val ownerLastName: String?,
    val sharedAt: Instant,
)
